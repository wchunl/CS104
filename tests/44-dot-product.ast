ROOT "" (0.0.0)
|   TOK_FUNCTION "" (6.22.1) function
|   |   TOK_TYPE_ID "" (6.22.1) typeid
|   |   |   TOK_VOID "void" (6.22.1) void
|   |   |   TOK_IDENT "fail" (6.22.6) variable
|   |   TOK_PARAM "(" (6.22.11) param
|   |   |   TOK_TYPE_ID "" (6.22.12) typeid
|   |   |   |   TOK_STRING "string" (6.22.12) string
|   |   |   |   TOK_IDENT "expr" (6.22.19) string variable (6.22.12)
|   |   |   TOK_TYPE_ID "" (6.22.25) typeid
|   |   |   |   TOK_STRING "string" (6.22.25) string
|   |   |   |   TOK_IDENT "file" (6.22.32) string variable (6.22.25)
|   |   |   TOK_TYPE_ID "" (6.22.38) typeid
|   |   |   |   TOK_INT "int" (6.22.38) int
|   |   |   |   TOK_IDENT "line" (6.22.42) int variable (6.22.38)
|   TOK_FUNCTION "" (6.24.1) function
|   |   TOK_TYPE_ID "" (6.24.1) typeid
|   |   |   TOK_VOID "void" (6.24.1) void
|   |   |   TOK_IDENT "putchr" (6.24.6) variable
|   |   TOK_PARAM "(" (6.24.13) param
|   |   |   TOK_TYPE_ID "" (6.24.14) typeid
|   |   |   |   TOK_INT "int" (6.24.14) int
|   |   |   |   TOK_IDENT "chr" (6.24.18) int variable (6.24.14)
|   TOK_FUNCTION "" (6.25.1) function
|   |   TOK_TYPE_ID "" (6.25.1) typeid
|   |   |   TOK_VOID "void" (6.25.1) void
|   |   |   TOK_IDENT "putint" (6.25.6) variable
|   |   TOK_PARAM "(" (6.25.13) param
|   |   |   TOK_TYPE_ID "" (6.25.14) typeid
|   |   |   |   TOK_INT "int" (6.25.14) int
|   |   |   |   TOK_IDENT "num" (6.25.18) int variable (6.25.14)
|   TOK_FUNCTION "" (6.26.1) function
|   |   TOK_TYPE_ID "" (6.26.1) typeid
|   |   |   TOK_VOID "void" (6.26.1) void
|   |   |   TOK_IDENT "putstr" (6.26.6) variable
|   |   TOK_PARAM "(" (6.26.13) param
|   |   |   TOK_TYPE_ID "" (6.26.14) typeid
|   |   |   |   TOK_STRING "string" (6.26.14) string
|   |   |   |   TOK_IDENT "str" (6.26.21) string variable (6.26.14)
|   TOK_FUNCTION "" (6.28.1) function
|   |   TOK_TYPE_ID "" (6.28.1) typeid
|   |   |   TOK_INT "int" (6.28.1) int
|   |   |   TOK_IDENT "getchr" (6.28.5) variable
|   |   TOK_PARAM "(" (6.28.11) param
|   TOK_FUNCTION "" (6.29.1) function
|   |   TOK_TYPE_ID "" (6.29.1) typeid
|   |   |   TOK_STRING "string" (6.29.1) string
|   |   |   TOK_IDENT "getstr" (6.29.8) variable
|   |   TOK_PARAM "(" (6.29.14) param
|   TOK_FUNCTION "" (6.30.1) function
|   |   TOK_TYPE_ID "" (6.30.1) typeid
|   |   |   TOK_STRING "string" (6.30.1) string
|   |   |   TOK_IDENT "getln" (6.30.8) variable
|   |   TOK_PARAM "(" (6.30.13) param
|   TOK_FUNCTION "" (7.5.1) function
|   |   TOK_TYPE_ID "" (7.5.1) typeid
|   |   |   TOK_INT "int" (7.5.1) int
|   |   |   TOK_IDENT "dot_product" (7.5.5) variable
|   |   TOK_PARAM "(" (7.5.17) param
|   |   |   TOK_TYPE_ID "" (7.5.18) typeid
|   |   |   |   TOK_INT "int" (7.5.18) int
|   |   |   |   TOK_IDENT "size" (7.5.22) int variable (7.5.18)
|   |   |   TOK_TYPE_ID "" (7.5.28) typeid
|   |   |   |   TOK_ARRAY "array" (7.5.28) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.5.34) typeid
|   |   |   |   TOK_IDENT "vec1" (7.5.39) array variable (7.5.28)
|   |   |   TOK_TYPE_ID "" (7.5.45) typeid
|   |   |   |   TOK_ARRAY "array" (7.5.45) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.5.51) typeid
|   |   |   |   TOK_IDENT "vec2" (7.5.56) array variable (7.5.45)
|   |   TOK_BLOCK "{" (7.5.62)
|   |   |   TOK_VARDECL "=" (7.6.14) variable lval
|   |   |   |   TOK_INT "int" (7.6.4) int
|   |   |   |   TOK_IDENT "index" (7.6.8) int variable (7.6.14)
|   |   |   |   TOK_INTCON "0" (7.6.16) int const
|   |   |   TOK_VARDECL "=" (7.7.12) variable lval
|   |   |   |   TOK_INT "int" (7.7.4) int
|   |   |   |   TOK_IDENT "dot" (7.7.8) int variable (7.7.12)
|   |   |   |   TOK_INTCON "0" (7.7.14) int const
|   |   |   TOK_WHILE "while" (7.8.4)
|   |   |   |   '<' "<" (7.8.17)
|   |   |   |   |   TOK_IDENT "index" (7.8.11) int variable (7.6.14)
|   |   |   |   |   TOK_IDENT "size" (7.8.19) int variable (7.5.18)
|   |   |   |   TOK_BLOCK "{" (7.8.25)
|   |   |   |   |   '=' "=" (7.9.11) int vreg
|   |   |   |   |   |   TOK_IDENT "dot" (7.9.7) int variable (7.7.12)
|   |   |   |   |   |   '+' "+" (7.9.17) int vreg
|   |   |   |   |   |   |   TOK_IDENT "dot" (7.9.13) int variable (7.7.12)
|   |   |   |   |   |   |   '*' "*" (7.9.31) int vreg
|   |   |   |   |   |   |   |   TOK_INDEX "[" (7.9.23) lval vaddr
|   |   |   |   |   |   |   |   |   TOK_IDENT "vec1" (7.9.19) array variable (7.5.28)
|   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.9.24) int variable (7.6.14)
|   |   |   |   |   |   |   |   TOK_INDEX "[" (7.9.37) lval vaddr
|   |   |   |   |   |   |   |   |   TOK_IDENT "vec2" (7.9.33) array variable (7.5.45)
|   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.9.38) int variable (7.6.14)
|   |   |   |   |   '=' "=" (7.10.13) int vreg
|   |   |   |   |   |   TOK_IDENT "index" (7.10.7) int variable (7.6.14)
|   |   |   |   |   |   '+' "+" (7.10.21) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.10.15) int variable (7.6.14)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.10.23) int const
|   |   |   TOK_RETURN "return" (7.12.4)
|   |   |   |   TOK_IDENT "dot" (7.12.11) int variable (7.7.12)
|   TOK_FUNCTION "" (7.17.1) function
|   |   TOK_TYPE_ID "" (7.17.1) typeid
|   |   |   TOK_INT "int" (7.17.1) int
|   |   |   TOK_IDENT "main" (7.17.5) variable
|   |   TOK_PARAM "(" (7.17.9) param
|   |   TOK_BLOCK "{" (7.17.12)
|   |   |   TOK_VARDECL "=" (7.18.20) variable lval
|   |   |   |   TOK_ARRAY "array" (7.18.4) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.18.10) typeid
|   |   |   |   TOK_IDENT "vec1" (7.18.15) array variable (7.5.28)
|   |   |   |   TOK_ALLOC "alloc" (7.18.22)
|   |   |   |   |   TOK_ARRAY "array" (7.18.28) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.18.34) typeid
|   |   |   |   |   TOK_INTCON "10" (7.18.41) int const
|   |   |   TOK_VARDECL "=" (7.19.20) variable lval
|   |   |   |   TOK_ARRAY "array" (7.19.4) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.19.10) typeid
|   |   |   |   TOK_IDENT "vec2" (7.19.15) array variable (7.5.45)
|   |   |   |   TOK_ALLOC "alloc" (7.19.22)
|   |   |   |   |   TOK_ARRAY "array" (7.19.28) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.19.34) typeid
|   |   |   |   |   TOK_INTCON "10" (7.19.41) int const
|   |   |   TOK_VARDECL "=" (7.20.14) variable lval
|   |   |   |   TOK_INT "int" (7.20.4) int
|   |   |   |   TOK_IDENT "index" (7.20.8) int variable (7.6.14)
|   |   |   |   TOK_INTCON "0" (7.20.16) int const
|   |   |   TOK_WHILE "while" (7.21.4)
|   |   |   |   '<' "<" (7.21.17)
|   |   |   |   |   TOK_IDENT "index" (7.21.11) int variable (7.6.14)
|   |   |   |   |   TOK_INTCON "10" (7.21.19) int const
|   |   |   |   TOK_BLOCK "{" (7.21.23)
|   |   |   |   |   '=' "=" (7.22.19) int vreg
|   |   |   |   |   |   TOK_INDEX "[" (7.22.11) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "vec1" (7.22.7) array variable (7.5.28)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.22.12) int variable (7.6.14)
|   |   |   |   |   |   '+' "+" (7.22.27) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.22.21) int variable (7.6.14)
|   |   |   |   |   |   |   TOK_INTCON "10" (7.22.29) int const
|   |   |   |   |   '=' "=" (7.23.19) int vreg
|   |   |   |   |   |   TOK_INDEX "[" (7.23.11) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "vec2" (7.23.7) array variable (7.5.45)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.23.12) int variable (7.6.14)
|   |   |   |   |   |   '*' "*" (7.23.27) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.23.21) int variable (7.6.14)
|   |   |   |   |   |   |   TOK_INTCON "10" (7.23.29) int const
|   |   |   |   |   '=' "=" (7.24.13) int vreg
|   |   |   |   |   |   TOK_IDENT "index" (7.24.7) int variable (7.6.14)
|   |   |   |   |   |   '+' "+" (7.24.21) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.24.15) int variable (7.6.14)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.24.23) int const
|   |   |   TOK_CALL "(" (7.26.11)
|   |   |   |   TOK_IDENT "putint" (7.26.4) variable
|   |   |   |   TOK_CALL "(" (7.26.24)
|   |   |   |   |   TOK_IDENT "dot_product" (7.26.12) variable
|   |   |   |   |   TOK_INTCON "10" (7.26.25) int const
|   |   |   |   |   |   TOK_IDENT "vec1" (7.26.29) array variable (7.5.28)
|   |   |   |   |   |   TOK_IDENT "vec2" (7.26.35) array variable (7.5.45)
|   |   |   TOK_CALL "(" (7.27.11)
|   |   |   |   TOK_IDENT "putchr" (7.27.4) variable
|   |   |   |   TOK_CHARCON "'\n'" (7.27.12) int const
|   |   |   TOK_RETURN "return" (7.28.4)
|   |   |   |   TOK_INTCON "0" (7.28.11) int const
