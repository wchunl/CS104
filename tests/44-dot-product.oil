fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
dot_product:.function int
          .param int size
          .param ptr vec1
          .param ptr vec2
          .local int index
          .local int dot
          index = 0
          dot = 0
.wh0:     $t0:i = index < size
          goto .od0 if not $t0:i
.do0:     $t1:p = vec2[index * :p]
          $t2:i = $t1:i * [
          $t3:i = $t2:i + *
          dot = $t3:i
          $t4:i = index + 1
          index = $t4:i
.od0:     return dot
          return 
          .end
main:     .function int
          .local ptr vec1
          .local ptr vec2
          .local int index
          malloc(4 * sizeof int)
          vec1 = $t5:i
          malloc(4 * sizeof int)
          vec2 = $t6:i
          index = 0
.wh1:     $t7:i = index < 10
          goto .od1 if not $t7:i
.do1:     $t8:i = index + 10
          vec1[index * :p] = $t8:i
          $t9:i = index * 10
          vec2[index * :p] = $t9:i
          $t10:i = index + 1
          index = $t10:i
.od1:     $t11:i = 10
          dot_product($t11:i)
          putint($t12:i)
          putchr('\n')
          return 0
          return 
          .end
