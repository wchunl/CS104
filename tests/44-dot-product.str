string_set[   0]:    4706206694200543916 0x2360de8->","
                   16579916199511856337 0x23624a8->"getln"
string_set[   1]:    6637313742931709005 0x2363d38->"*"
                    2297668033614959926 0x23633c8->"0"
                    9330387579099533542 0x2363078->"vec2"
string_set[   3]:   11868120863286193964 0x2364398->"main"
                    9611428142580219387 0x235c6f8->"fail"
string_set[   4]:      21606623943142546 0x2362e88->"{"
                    4017852018407011111 0x2360ef8->"file"
string_set[   6]:   13686721134187162920 0x2363f58->"}"
string_set[   7]:   11374676767421005072 0x2362718->"dot_product"
string_set[   8]:    1370730858159036685 0x2363698->"+"
                    9386170080689500517 0x2360c98->"expr"
string_set[   9]:   10319632053171904911 0x2363498->"while"
                   14526079891305879776 0x2362df8->">"
string_set[  12]:   15816506932648033945 0x235c328->"("
string_set[  14]:   15298581498043935097 0x2363478->"dot"
string_set[  17]:   18413295556186398326 0x23612e8->"putint"
string_set[  18]:   13934964484049229816 0x23629c8->"size"
                    7211130878751977441 0x2361438->"putchr"
                   12679825168387489523 0x23619c8->"num"
string_set[  21]:   13581035385482130682 0x2363c08->"["
string_set[  23]:   12051031859636400188 0x2363d18->"]"
                   11166701219533395393 0x2362218->"getstr"
                   12868445110721718657 0x2360f18->")"
string_set[  28]:   14635384904416252381 0x23636b8->"return"
string_set[  29]:    2594088158685378223 0x2364b28->"10"
                   12291392672316406109 0x23610d8->"line"
string_set[  31]:    6192799629375248351 0x23656b8->"'\n'"
                    6142509188972423790 0x235c6d8->""
                   13348923631108115385 0x2361758->"putstr"
string_set[  32]:   11597697714117577063 0x2363318->"="
string_set[  33]:    8290819294854170237 0x23647d8->"alloc"
                    1583576014053361350 0x2361b78->"str"
string_set[  38]:   12597418120160561823 0x2360d18->"int"
string_set[  39]:    7296690267450279874 0x23615a8->"chr"
string_set[  40]:   11911796782241914524 0x23607d8->"void"
string_set[  42]:    4698864730501174006 0x2363098->"index"
                    5344662657899890615 0x23610f8->";"
string_set[  43]:    6814028903552918164 0x2361298->"array"
string_set[  44]:   13497578857207247412 0x2360bb8->"string"
string_set[  45]:   10159970873491820195 0x2364178->"1"
                    1396094175042499165 0x2362ce8->"<"
string_set[  46]:    2486443799175530694 0x2362e68->"vec1"
                    9201800968376702500 0x2361fa8->"getchr"
load_factor = 0.957
bucket_count = 47
max_bucket_size = 3
