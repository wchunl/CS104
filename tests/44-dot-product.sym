fail (6.22.1) {0} void function
   expr (6.22.12) {1} string variable param lval 0
   file (6.22.25) {1} string variable param lval 1
   line (6.22.38) {1} int variable param lval 2

putchr (6.24.1) {1} void function
   chr (6.24.14) {1} int variable param lval 0

putint (6.25.1) {2} void function
   num (6.25.14) {1} int variable param lval 0

putstr (6.26.1) {3} void function
   str (6.26.14) {1} string variable param lval 0

getchr (6.28.1) {4} int function

getstr (6.29.1) {5} string function

getln (6.30.1) {6} string function

dot_product (7.5.1) {7} int function
   size (7.5.18) {1} int variable param lval 0
   vec1 (7.5.28) {1} array <int> variable param lval 1
   vec2 (7.5.45) {1} array <int> variable param lval 2
   index (7.6.14) {1} int variable local lval 0
   dot (7.7.12) {1} int variable local lval 1

main (7.17.1) {8} int function
   vec1 (7.18.20) {1} array <int> variable local lval 0
   vec2 (7.19.20) {1} array <int> variable local lval 1
   index (7.20.14) {1} int variable local lval 2
