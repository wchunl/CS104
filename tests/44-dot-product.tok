#  1 "44-dot-product.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "44-dot-product.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  4 "44-dot-product.oc"
   7 5.00001 262 TOK_INT       int
   7 5.00005 282 TOK_IDENT     dot_product
   7 5.00017  40 '('           (
   7 5.00018 262 TOK_INT       int
   7 5.00022 282 TOK_IDENT     size
   7 5.00026  44 ','           ,
   7 5.00028 271 TOK_ARRAY     array
   7 5.00033  60 '<'           <
   7 5.00034 262 TOK_INT       int
   7 5.00037  62 '>'           >
   7 5.00039 282 TOK_IDENT     vec1
   7 5.00043  44 ','           ,
   7 5.00045 271 TOK_ARRAY     array
   7 5.00050  60 '<'           <
   7 5.00051 262 TOK_INT       int
   7 5.00054  62 '>'           >
   7 5.00056 282 TOK_IDENT     vec2
   7 5.00060  41 ')'           )
   7 5.00062 123 '{'           {
   7 6.00004 262 TOK_INT       int
   7 6.00008 282 TOK_IDENT     index
   7 6.00014  61 '='           =
   7 6.00016 283 TOK_INTCON    0
   7 6.00017  59 ';'           ;
   7 7.00004 262 TOK_INT       int
   7 7.00008 282 TOK_IDENT     dot
   7 7.00012  61 '='           =
   7 7.00014 283 TOK_INTCON    0
   7 7.00015  59 ';'           ;
   7 8.00004 267 TOK_WHILE     while
   7 8.00010  40 '('           (
   7 8.00011 282 TOK_IDENT     index
   7 8.00017  60 '<'           <
   7 8.00019 282 TOK_IDENT     size
   7 8.00023  41 ')'           )
   7 8.00025 123 '{'           {
   7 9.00007 282 TOK_IDENT     dot
   7 9.00011  61 '='           =
   7 9.00013 282 TOK_IDENT     dot
   7 9.00017  43 '+'           +
   7 9.00019 282 TOK_IDENT     vec1
   7 9.00023  91 '['           [
   7 9.00024 282 TOK_IDENT     index
   7 9.00029  93 ']'           ]
   7 9.00031  42 '*'           *
   7 9.00033 282 TOK_IDENT     vec2
   7 9.00037  91 '['           [
   7 9.00038 282 TOK_IDENT     index
   7 9.00043  93 ']'           ]
   7 9.00044  59 ';'           ;
   7 10.00007 282 TOK_IDENT     index
   7 10.00013  61 '='           =
   7 10.00015 282 TOK_IDENT     index
   7 10.00021  43 '+'           +
   7 10.00023 283 TOK_INTCON    1
   7 10.00024  59 ';'           ;
   7 11.00004 125 '}'           }
   7 12.00004 268 TOK_RETURN    return
   7 12.00011 282 TOK_IDENT     dot
   7 12.00014  59 ';'           ;
   7 13.00001 125 '}'           }
   7 17.00001 262 TOK_INT       int
   7 17.00005 282 TOK_IDENT     main
   7 17.00009  40 '('           (
   7 17.00010  41 ')'           )
   7 17.00012 123 '{'           {
   7 18.00004 271 TOK_ARRAY     array
   7 18.00009  60 '<'           <
   7 18.00010 262 TOK_INT       int
   7 18.00013  62 '>'           >
   7 18.00015 282 TOK_IDENT     vec1
   7 18.00020  61 '='           =
   7 18.00022 273 TOK_ALLOC     alloc
   7 18.00027  60 '<'           <
   7 18.00028 271 TOK_ARRAY     array
   7 18.00033  60 '<'           <
   7 18.00034 262 TOK_INT       int
   7 18.00037  62 '>'           >
   7 18.00038  62 '>'           >
   7 18.00040  40 '('           (
   7 18.00041 283 TOK_INTCON    10
   7 18.00043  41 ')'           )
   7 18.00044  59 ';'           ;
   7 19.00004 271 TOK_ARRAY     array
   7 19.00009  60 '<'           <
   7 19.00010 262 TOK_INT       int
   7 19.00013  62 '>'           >
   7 19.00015 282 TOK_IDENT     vec2
   7 19.00020  61 '='           =
   7 19.00022 273 TOK_ALLOC     alloc
   7 19.00027  60 '<'           <
   7 19.00028 271 TOK_ARRAY     array
   7 19.00033  60 '<'           <
   7 19.00034 262 TOK_INT       int
   7 19.00037  62 '>'           >
   7 19.00038  62 '>'           >
   7 19.00040  40 '('           (
   7 19.00041 283 TOK_INTCON    10
   7 19.00043  41 ')'           )
   7 19.00044  59 ';'           ;
   7 20.00004 262 TOK_INT       int
   7 20.00008 282 TOK_IDENT     index
   7 20.00014  61 '='           =
   7 20.00016 283 TOK_INTCON    0
   7 20.00017  59 ';'           ;
   7 21.00004 267 TOK_WHILE     while
   7 21.00010  40 '('           (
   7 21.00011 282 TOK_IDENT     index
   7 21.00017  60 '<'           <
   7 21.00019 283 TOK_INTCON    10
   7 21.00021  41 ')'           )
   7 21.00023 123 '{'           {
   7 22.00007 282 TOK_IDENT     vec1
   7 22.00011  91 '['           [
   7 22.00012 282 TOK_IDENT     index
   7 22.00017  93 ']'           ]
   7 22.00019  61 '='           =
   7 22.00021 282 TOK_IDENT     index
   7 22.00027  43 '+'           +
   7 22.00029 283 TOK_INTCON    10
   7 22.00031  59 ';'           ;
   7 23.00007 282 TOK_IDENT     vec2
   7 23.00011  91 '['           [
   7 23.00012 282 TOK_IDENT     index
   7 23.00017  93 ']'           ]
   7 23.00019  61 '='           =
   7 23.00021 282 TOK_IDENT     index
   7 23.00027  42 '*'           *
   7 23.00029 283 TOK_INTCON    10
   7 23.00031  59 ';'           ;
   7 24.00007 282 TOK_IDENT     index
   7 24.00013  61 '='           =
   7 24.00015 282 TOK_IDENT     index
   7 24.00021  43 '+'           +
   7 24.00023 283 TOK_INTCON    1
   7 24.00024  59 ';'           ;
   7 25.00004 125 '}'           }
   7 26.00004 282 TOK_IDENT     putint
   7 26.00011  40 '('           (
   7 26.00012 282 TOK_IDENT     dot_product
   7 26.00024  40 '('           (
   7 26.00025 283 TOK_INTCON    10
   7 26.00027  44 ','           ,
   7 26.00029 282 TOK_IDENT     vec1
   7 26.00033  44 ','           ,
   7 26.00035 282 TOK_IDENT     vec2
   7 26.00039  41 ')'           )
   7 26.00040  41 ')'           )
   7 26.00041  59 ';'           ;
   7 27.00004 282 TOK_IDENT     putchr
   7 27.00011  40 '('           (
   7 27.00012 284 TOK_CHARCON   '\n'
   7 27.00016  41 ')'           )
   7 27.00017  59 ';'           ;
   7 28.00004 268 TOK_RETURN    return
   7 28.00011 283 TOK_INTCON    0
   7 28.00012  59 ';'           ;
   7 29.00001 125 '}'           }
