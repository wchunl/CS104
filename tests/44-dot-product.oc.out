; 0.  1: # 1 "44-dot-product.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "44-dot-product.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 4 "44-dot-product.oc" 2
; 7.  4: 
; 7.  5: int dot_product (int size, array<int> vec1, array<int> vec2) {
; 7.  6:    int index = 0;
; 7.  7:    int dot = 0;
; 7.  8:    while (index < size) {
; 7.  9:       dot = dot + vec1[index] * vec2[index];
; 7. 10:       index = index + 1;
; 7. 11:    }
; 7. 12:    return dot;
; 7. 13: }
; 7. 14: 
; 7. 15: 
; 7. 16: 
; 7. 17: int main() {
; 7. 18:    array<int> vec1 = alloc<array<int>> (10);
; 7. 19:    array<int> vec2 = alloc<array<int>> (10);
; 7. 20:    int index = 0;
; 7. 21:    while (index < 10) {
; 7. 22:       vec1[index] = index + 10;
; 7. 23:       vec2[index] = index * 10;
; 7. 24:       index = index + 1;
; 7. 25:    }
; 7. 26:    putint (dot_product (10, vec1, vec2));
; 7. 27:    putchr ('\n');
; 7. 28:    return 0;
; 7. 29: }
